precise/iaf_psc_alpha_ps.cpp

```diff
@@ -578,36 +575,78 @@ nest::iaf_psc_alpha_ps::emit_instant_spike_( Time const& origin, const long lag,
 }
 
 double
-nest::iaf_psc_alpha_ps::bisectioning_( const double dt ) const
+nest::iaf_psc_alpha_ps::V_m_root_function_( double t_step ) const
 {
-  double root = 0.0;
+  const double ps_P30 = -P_.tau_m_ / P_.c_m_ * numerics::expm1( -t_step / P_.tau_m_ );
+
+  const double ps_P31_ex = propagator_31( P_.tau_syn_ex_, P_.tau_m_, P_.c_m_, t_step );
+  const double ps_P32_ex = propagator_32( P_.tau_syn_ex_, P_.tau_m_, P_.c_m_, t_step );
+  const double ps_P31_in = propagator_31( P_.tau_syn_in_, P_.tau_m_, P_.c_m_, t_step );
+  const double ps_P32_in = propagator_32( P_.tau_syn_in_, P_.tau_m_, P_.c_m_, t_step );
+
+  double V_m_root = ps_P30 * ( P_.I_e_ + V_.y_input_before_ ) + ps_P31_ex * V_.dI_ex_before_
+    + ps_P32_ex * V_.I_ex_before_ + ps_P31_in * V_.dI_in_before_ + ps_P32_in * V_.I_in_before_
+    + V_.V_m_before_ * std::exp( -t_step / P_.tau_m_ );
+
+  return V_m_root - P_.U_th_;
+}
+
+double
+nest::iaf_psc_alpha_ps::regula_falsi_method_( const double dt ) const
+{
+  double root;
+  double V_m_root;
+
+  int side = 0;
+
+  double a_k = 0.0;
+  double b_k = dt;
+
+  double V_m_func_a_k = V_m_root_function_( a_k );
+  double V_m_func_b_k = V_m_root_function_( b_k );
 
-  double V_m_root = V_.V_m_before_;
+  assert( V_m_func_a_k * V_m_func_b_k <= 0 );
 
-  double div = 2.0;
+  int max_iter = 500;
 
-  while ( fabs( P_.U_th_ - V_m_root ) > 1e-14 )
+  for ( int iter = 0; iter < max_iter; ++iter )
   {
-    if ( V_m_root > P_.U_th_ )
-    {
-      root -= dt / div;
-    }
-    else
+    root = ( a_k * V_m_func_b_k - b_k * V_m_func_a_k ) / ( V_m_func_b_k - V_m_func_a_k );
+    V_m_root = V_m_root_function_( root );
+
+    if ( std::abs( V_m_root ) <= 1e-14 )
```

Comment:
I would avoid `<=` if possible, a simple `<` is cleaner and has the same effect here. The limit value should be defined as named constant.